language-guides/spring-data-neo4j/spring-data-neo4j.adoc

```diff
@@ -34,129 +34,206 @@ When developing with Neo4j, please use JDK 8 and your favorite IDE.
+The Role class also has an ID field that is generated by the database and an array list to contain a possible list of roles (person could play more than one role in a movie).
+Finally, annotations are added to mark the Person and Movie nodes as start and end nodes for the relationship.
+We now have our graph structure mapped in our application. This is the object-graph mapping (OGM) piece.
 
 .Entities
 [source,java]
 ----
 @NodeEntity
 public class Movie {
 
-  @Id @GeneratedValue Long id;
+    @Id @GeneratedValue private Long id;
+	private String title;
+	private int released;
+	private String tagline;
 
-  String title;
+	@JsonIgnoreProperties("movie")
+	@Relationship(type = "ACTED_IN", direction = Relationship.INCOMING)
+	private List<Role> roles;
+}
+
+@NodeEntity
+public class Person {
 
-  Person director;
+    @Id @GeneratedValue private Long id;
+	private String name;
+	private int born;
 
-  @Relationship(type="ACTED_IN", direction = Relationship.INCOMING)
-  Set<Person> actors = new HashSet<>();
+	@Relationship(type = "ACTED_IN")
+	private List<Movie> movies = new ArrayList<>();
 }
+
+@RelationshipEntity(type = "ACTED_IN")
+public class Role {
+
+    @Id @GeneratedValue private Long id;
+	private List<String> roles = new ArrayList<>();
+
+	@StartNode
+	private Person person;
+
+	@EndNode
+	private Movie movie;
 ----
 
+The repository interface allows the developer to create methods and queries needed for retrieving the data from the database.
+The interface extends the Neo4jRepository, which extends the Spring CRUDRepository for persisting and retrieving data.
+Some basic queries are derived for us (like the findByTitleLike method), but others must be specifically written and annotated with @Query.
+
 .Declare a repository interface
 [source,java]
 ----
-interface MovieRepository extends Neo4jRepository<Movie, Long> {
+public interface MovieRepository extends Neo4jRepository<Movie, Long> {
+
+    @Query("MATCH (m:Movie {title: {title}}) RETURN m LIMIT 1")
+	Movie findByTitle(@Param("title") String title);
 
-  @Query("MATCH (m:Movie)<-[rating:RATED]-(user)
-          WHERE id(m) = {movieId} RETURN rating")
-  List<Rating> getRatings(@Param("movieID") Long movieId);
+	Collection<Movie> findByTitleLike(@Param("title") String title);
 
-  List<Movie> findByTitle(String title);
+    @Query("MATCH (m:Movie)<-[r:ACTED_IN]-(a:Person) RETURN m,r,a LIMIT {limit}")
+	Collection<Movie> graph(@Param("limit") int limit);
 }
 ----
 
+In the code below, the MovieService class injects the MovieRepository in using a constructor.
+The next blocks of code are the implementations for the methods defined in the MovieRepository class.
+Annotating these methods with @Transactional and read-only will allow enterprise applications using causal clustering to improve performance by routing these transactions to replica servers and focus the write traffic to the core servers.
```

Comment:
Not sure if this is a bit too involved with details for the "quickstart"?
Perhaps just mention that the smart client can use that information for efficient routing & execution.